lib/queuePopulator/IngestionReader.js

```diff
@@ -71,22 +71,35 @@ class IngestionReader extends LogReader {
     }
 
     setup(done) {
-        this._setupIngestionProducer(err => {
-            if (err) {
-                return done(err);
-            }
-            return this._iProducer.getRaftId(this.bucket, (err, data) => {
+        const retryableError = { retryable: true };
+        async.series([
+            next => this._setupIngestionProducer(next),
+            next => this._iProducer.validateSourceSetup(this.bucket,
+                this.logger, err => {
+                    // these errors are retryable and should not crash pod
+                    if (err) {
+                        return next(retryableError);
+                    }
+                    return next();
+                }),
+            next => this._iProducer.getRaftId(this.bucket, (err, data) => {
                 if (err) {
-                    return done(err);
+                    if (err.code === 'AccessDenied') {
+                        // retryable
+                        this.logger.error('admin credentials are required', {
+                            error: err,
+                        });
+                        return next(retryableError);
+                    }
+                    return next(err);
```

Comment:
same here, if an error occurred and it's not an 'AccessDenied' error, I crash the pod.

The potential error here could be that the source location is down and inaccessible.